### PR TITLE
T26806 Fix tests running locally

### PR DESCRIPTION
**Pull request by mwleeds**
https://phabricator.endlessm.com/T26806

### CODE REVIEW

File: test-common/utils.c [hunk 1]
```diff
@@ -3002,9 +3002,8 @@ eos_test_client_prepare_volume (EosTestClient *client,
       { "GI_TYPELIB_PATH", libeos_updater_util_path, NULL },
       { "LD_LIBRARY_PATH", new_ld_library_path, NULL },
       { "FLATPAK_SYSTEM_HELPER_ON_SESSION", "1", NULL },
-      /* FIXME: Add back G_DEBUG=fatal-warnings after we can rely on
-       * https://gitlab.gnome.org/GNOME/pygobject/commit/806c5059f989ed1b8bc62e6aa1ef55123ac110de */
-      { "G_DEBUG", "gc-friendly", NULL },
+      { "XDG_CACHE_HOME", g_get_user_cache_dir (), NULL },
```
Comment (pwithnall) [inline test-common/utils.c]: Would this not come from the parent process’ environment through `merge_parent_and_child_env()`?

Comment (mwleeds) [reply test-common/utils.c]: No because the parent process has `XDG_CACHE_HOME=/dev/null` because of `G_TEST_OPTION_ISOLATE_DIRS`

Comment (pwithnall) [reply test-common/utils.c]: Worth a comment in the code.

Comment (mwleeds) [reply test-common/utils.c]: Oh yeah, I had a comment in a previous version, guess I didn't copy it over

File: Makefile.am [hunk 2]
```diff
@@ -631,6 +631,14 @@ test_common_libeos_updater_test_common_la_SOURCES = \
 	test-common/utils.h \
 	$(NULL)
 
+tests_services_in = \
+	tests/services/org.freedesktop.Flatpak.SystemHelper.service.in \
+	$(NULL)
+
+systemdsystemunit_DATA += $(tests_services_in:%.in=%)
```
Comment (pwithnall) [inline Makefile.am]: Not sure the `systemdsystemunit_DATA` variable is the right one for this; that means this mock D-Bus service file will be installed system-wide in the systemd system unit directory.

File: tests/services/org.freedesktop.Flatpak.SystemHelper.service.in [hunk 3]
```diff
@@ -0,0 +1,5 @@
+[D-BUS Service]
+Name=org.freedesktop.Flatpak.SystemHelper
+Exec=@libexecdir@/flatpak-system-helper --session --no-idle-exit
```
Comment (pwithnall) [inline tests/services/org.freedesktop.Flatpak.SystemHelper.service.in]: Might be worth adding a comment here to point out that `--session` is added deliberately to make it work with our mock D-Bus session bus.

File: Makefile.am (hunk 2 above)
Comment (mwleeds) [reply Makefile.am]: Ah right that was wrong on two levels: it's not a systemd unit and it shouldn't be installed. Pushed a fixup to put it in `noinst_DATA` and verified that it doesn't get installed but the tests still pass. Hopefully that's right; autotools are still a mystery to me

File: tests/services/org.freedesktop.Flatpak.SystemHelper.service.in (hunk 3 above)
Comment (mwleeds) [reply tests/services/org.freedesktop.Flatpak.SystemHelper.service.in]: done

File: Makefile.am (hunk 2 above)
Comment (pwithnall) [reply Makefile.am]: Looks good, although I realise I have just merged the fixups unsquashed. :-(

**Pull request merged**